checks.d/hdfs_namenode.py

```diff
@@ -0,0 +1,193 @@
+hdfs.namenode.num_stale_storages                Number of stale storages
+'''
+
+# stdlib
+from urlparse import urljoin
+
+# 3rd party
+import requests
+from requests.exceptions import Timeout, HTTPError, InvalidURL, ConnectionError
+from simplejson import JSONDecodeError
+
+# Project
+from checks import AgentCheck
+
+# JMX URI
+JMX_URI = 'http://localhost:50070'
+
+# Service check names
+JMX_SERVICE_CHECK = 'hdfs.namenode.jmx.can_connect'
+
+# URL Paths
+JMX_PATH = 'jmx'
+
+# Namenode Bean
+HDFS_NAMENODE_BEAN_NAME = 'Hadoop:service=NameNode,name=FSNamesystemState'
+
+# Metric types
+GAUGE = 'gauge'
+
+# HDFS metrics
+HDFS_METRICS = {
+    'CapacityTotal' : ('hdfs.namenode.capacity_total',  GAUGE),
+    'CapacityUsed' : ('hdfs.namenode.capacity_used',  GAUGE),
+    'CapacityRemaining' : ('hdfs.namenode.capacity_remaining',  GAUGE),
+    'TotalLoad' : ('hdfs.namenode.total_load',  GAUGE),
+    'FsLockQueueLength' : ('hdfs.namenode.fs_lock_queue_length',  GAUGE),
+    'BlocksTotal' : ('hdfs.namenode.blocks_total',  GAUGE),
+    'MaxObjects' : ('hdfs.namenode.max_objects',  GAUGE),
+    'FilesTotal' : ('hdfs.namenode.files_total',  GAUGE),
+    'PendingReplicationBlocks' : ('hdfs.namenode.pending_replication_blocks',  GAUGE),
+    'UnderReplicatedBlocks' : ('hdfs.namenode.under_replicated_blocks',  GAUGE),
+    'ScheduledReplicationBlocks' : ('hdfs.namenode.scheduled_replication_blocks',  GAUGE),
+    'PendingDeletionBlocks' : ('hdfs.namenode.pending_deletion_blocks',  GAUGE),
+    'NumLiveDataNodes' : ('hdfs.namenode.num_live_data_nodes',  GAUGE),
+    'NumDeadDataNodes' : ('hdfs.namenode.num_dead_data_nodes',  GAUGE),
+    'NumDecomLiveDataNodes' : ('hdfs.namenode.num_decom_live_data_nodes',  GAUGE),
+    'NumDecomDeadDataNodes' : ('hdfs.namenode.num_decom_dead_data_nodes',  GAUGE),
+    'VolumeFailuresTotal' : ('hdfs.namenode.volume_failures_total',  GAUGE),
+    'EstimatedCapacityLostTotal' : ('hdfs.namenode.estimated_capacity_lost_total',  GAUGE),
+    'NumDecommissioningDataNodes' : ('hdfs.namenode.num_decommissioning_data_nodes',  GAUGE),
+    'NumStaleDataNodes' : ('hdfs.namenode.num_stale_data_nodes',  GAUGE),
+    'NumStaleStorages' : ('hdfs.namenode.num_stale_storages',  GAUGE),
+}
+
+class HDFSNameNode(AgentCheck):
+
+    def check(self, instance):
+        jmx_address = instance.get('hdfs_namenode_jmx_uri', JMX_URI)
+
+        # Get metrics from JMX
+        self._hdfs_namenode_metrics(jmx_address, JMX_PATH)
+
+    def _hdfs_namenode_metrics(self, jmx_uri, jmx_path):
+        '''
+        Get HDFS data node metrics from JMX
+        '''
+        response = self._rest_request_to_json(jmx_uri,
+            jmx_path,
+            JMX_SERVICE_CHECK,
+            qry='Hadoop:service=NameNode,name=FSNamesystemState')
+
+        beans = response.get('beans', [])
+
+        for bean in beans:
+            bean_name = bean.get('name')
+
+            if bean_name is not None \
```

Comment:
no need to check that `bean_name` is not `None` here as we already compare it with `HDFS_NAMENODE_BEAN_NAME` ;)